standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -815,26 +855,30 @@ List<Table> getTables(String catName, String dbName, List<String> tableNames, Ge
    * @throws MetaException
    *          Any other errors
    */
-  List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
-      throws MetaException, InvalidOperationException, UnknownDBException, TException;
+  default List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
+      throws MetaException, InvalidOperationException, UnknownDBException, TException{
+    return Collections.emptyList();
+  }
 
   /**
    * Returns the invalidation information for the materialized views given as input.
    */
-  Materialization getMaterializationInvalidationInfo(CreationMetadata cm, String validTxnList)
-          throws MetaException, InvalidOperationException, UnknownDBException, TException;
+  default Materialization getMaterializationInvalidationInfo(CreationMetadata cm, String validTxnList)
+          throws MetaException, InvalidOperationException, UnknownDBException, TException{
+    return new Materialization();
+  }
 
   /**
    * Updates the creation metadata for the materialized view.
    */
-  void updateCreationMetadata(String dbName, String tableName, CreationMetadata cm)
-      throws MetaException, TException;
+  default void updateCreationMetadata(String dbName, String tableName, CreationMetadata cm)
```

Comment:
should throw UnsupportedOperationException

Reply:
Done.